web_src/js/components/RepoActionView.vue

```diff
@@ -186,10 +189,20 @@ const sfc = {
     },
 
     // show/hide the step logs for a step
-    toggleStepLogs(idx) {
-      this.currentJobStepsStates[idx].expanded = !this.currentJobStepsStates[idx].expanded;
-      if (this.currentJobStepsStates[idx].expanded) {
-        this.loadJob(); // try to load the data immediately instead of waiting for next timer interval
+    async toggleStepLogs(idx) {
+      const currentToggledStep = this.currentJobStepsStates[idx];
+      currentToggledStep.expanded = !currentToggledStep.expanded;
+      if (currentToggledStep.expanded) {
+        // If the job is done and the job step log is expanded for the first time,
+        // wait for fetching job and show the loading icon
+        if (this.isDone(this.run.status) && currentToggledStep.isInitialExpand) {
+          currentToggledStep.isInitialExpand = false;
+          currentToggledStep.isLoading = true;
+          await this.loadJob();
+          currentToggledStep.isLoading = false;
+        } else {
+          this.loadJob(); // try to load the data immediately instead of waiting for next timer interval
+        }
```

Comment:
How about simplifying it to : `expanded && no-record` means loading?

Reply:
What is `no-record(no-cursor)`, Is this a function?

Reply:
```
            this.currentJobStepsStates[i] = {cursor: null, expanded: false};
```

Reply:
I see, will change this later, thanks.